stream/BidirectionalStream.java

```diff
@@ -118,22 +118,40 @@ private void close(@Nullable StatusRuntimeException error) {
         }
     }
 
-    public List<StatusRuntimeException> drainErrors() {
-        List<StatusRuntimeException> errors = new ArrayList<>(resCollector.drainErrors());
-        errors.addAll(resPartCollector.drainErrors());
+    void singleDone(UUID requestID) {
+        resCollector.remove(requestID);
+    }
+
+    void iteratorDone(UUID requestID) {
+        resPartCollector.remove(requestID);
+    }
+
+    public List<StatusRuntimeException> getErrors() {
+        List<StatusRuntimeException> errors = new ArrayList<>(resCollector.getErrors());
+        errors.addAll(resPartCollector.getErrors());
         return errors;
     }
 
+    RequestTransmitter.Dispatcher dispatcher() {
+        return dispatcher;
+    }
```

Comment:
we've replaced the `ResPartIterator` argument which is `dispatcher` with a getter for `dispatcher()` on the stream, which the Iterator now has access to